### ISSUE TITLE
Memory leak in BMS

### ISSUE THREAD

**Issue by tiamat2012** [crowbarmaster/BedrockManagementService#82]
When my Minecraft server hangs, I noticed that the amount of memory being consumed by the service is quite significant.

![IMG_0376](https://user-images.githubusercontent.com/16579793/171760516-21b075b0-82ac-4e08-8fed-332b6410f388.jpeg)

On a clean start of the same world, the service is using around 27 MB of memory.

**Comment by crowbarmaster** [comment]
Ouch! I managed to run some tests on this. With two servers running and zero interaction with the service, there are no issues. However things seemed to jump way high when I saved service settings, causing the service to "reboot". This shot mem usage up to ~550MB. Something is off here for sure, and I will do some digging into it!

**Comment by tiamat2012** [comment]
![IMG_0414](https://user-images.githubusercontent.com/16579793/173734083-58bb23a4-6e13-4cf4-8cb4-69f64c8a01e2.jpeg)

We reached an impressive 2.7GB for the service today before I killed and restarted it.

I am only running 1 world on one server/service.  But I do have a lot of backups (I backup every 30 minutes while people are playing during the evening or weekend).  So I bet the leak has something to do with the backup process.

**Comment by crowbarmaster** [comment]
Alrighty, sounds like it's possible the zip archives of the backups are not getting disposed of as they should. I should have time to look into it this weekend. I will be back in touch soon!

**Comment by crowbarmaster** [comment]
I did find something in the auto-updater that would cause a rouge loop, even seemed to happen with updates off and version locked. Also it would seem that restarts from the servers would stack up backup timers, as I had missed a call to stop them. I was unable to find the time to verify this caused a leak, but it feels right, lol. Let me know how this build performs! @tiamat2012

**Comment by tiamat2012** [comment]
@crowbarmaster Thanks for the update!

I updated today and will see how it goes with the mem leak and backup count max. Just before I updated, the BMS services was just under 4.4 GB of memory usage.

The autobackup has recently saved me twice! I have had 2 power outages and even though there has been no one playing on the server at the time, BDS loses its mind and dumps the entire world into a "lost" folder and the world starts blank.  I go back to a backup file and restore the world folder and I am back up in minutes, and didn't lose anything because of the way the backup system works and saves after the last person disconnects.  BDS seems to be a lot more sensitive these days and holding things in memory that can cause the world database to get corrupted.
